### ISSUE TITLE
Identify joint and link parameters based on the real robot

### ISSUE THREAD

**Issue by zichunxx** [lvjonok/mujoco-sysid#10]
Hi! Thanks for your generous work, which benefits me a lot.

I'm new to system identification. If it would be considered that updating some tutorials about identifying joint friction or link inertial parameters based on the real robot? It would be better to finetune related parameters in the MuJoCo XML file.

Thanks!

**Comment by lvjonok** [comment]
Hello @zichunxx!

Indeed, it is a very interesting showcase for system identification and further update of MuJoCo XML file. I have been working for a while on the problem you are talking about, if you are willing to share log data from the real robot, I will create a new tutorial.

Looking forward for your response!

**Comment by zichunxx** [comment]
Thanks for your reply. Are joint position, velocity, acceleration, torque, and the cartesian pose of the end-effector all the data needed when following an excitation trajectory? I have a KUKA IIWA 14 robot and can try to collect some log data for your reference. However, I'm not sure if a special design is needed for the excitation trajectory in this scenario.

**Comment by lvjonok** [comment]
> Thanks for your reply. Are joint position, velocity, acceleration, torque, and the cartesian pose of the end-effector all the data needed when following an excitation trajectory? I have a KUKA IIWA 14 robot and can try to collect some log data for your reference. However, I'm not sure if a special design is needed for the excitation trajectory in this scenario.

Yes, excitation of trajectory will affect the quality of identification, however I have in mind some other approaches that can reduce the dependency on trajectory quality. I think the trajectory needed should contain timestamps, joint position, velocity, (possibly acceleration) and torque.

**Comment by zichunxx** [comment]
Okay, I'll try to reproduce some methods to design the excitation trajectory. Any update will be shared here.

**Comment by lvjonok** [comment]
> Okay, I'll try to reproduce some methods to design the excitation trajectory. Any update will be shared here. 

Great! Let's do something good!

**Comment by lvjonok** [comment]
> Hi, @lvjonok. Sorry for the late reply.
> 
> I designed an excitation trajectory and logged some data with KUKA IIWA 14 LBR R820.
> 
> Here is the [data.csv](https://github.com/user-attachments/files/18034367/data.csv), where `mp*` and `mt*` stand for the measured position and torque of different joints. The data is sent and collected at 500Hz. However, the designed excitation trajectory may not be perfect. If so, you can provide a new trajectory for me and I'll log the data again.
> 
> Looking forward to the new update!

Hello, @zichunxx !

I very much appreciate your effort! Right now I am lacking time due my finals approaching, so I will get to this in around 2 weeks.

**Comment by rris-Wyf** [comment]
Hi, @lvjonok.
Thanks for your excellent work which inspires and helps me a lot.
I am trying to use the` joint_torque_regressor` and I noticed that you used `mujoco.mj_objectVelocity` and `mujoco.mj_objectAcceleration` to get the body velocities and accelerations.
But if I set the `mj_data.qpos`, `mj_data.qvel` and `mj_data.qacc` with my own data, it must be some actions to let mujoco update the body velocities and accelerations. **Which function have you called in your testing**?
`mj_forward` and ‘mj_step’ doesn't look correct to me
